src/Microsoft.Azure.SignalR.Common/ServiceConnections/ServiceConnectionScopeInternal.cs

```diff
@@ -0,0 +1,63 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Diagnostics;
+using System.Threading;
+
+namespace Microsoft.Azure.SignalR
+{
+    internal class ClientConnectionScopeProperties
+    {
+        public IServiceConnection ServiceConnection { get; set; }
+
+        // todo: add additional properties to flow with the scope here
+    }
+
+    internal class ServiceConnectionScopeHolder
+    {
+        private static AsyncLocal<ServiceConnectionScopeHolder> s_AsyncLocalSCC = new AsyncLocal<ServiceConnectionScopeHolder>();
```

Comment:
nit: naming: `_currentScope`?

Reply:
Agree, the current name is weird. I do want the name to signify that it is async local and static though.

Reply:
add `readonly`